heudiconv/convert.py

```diff
@@ -80,7 +80,7 @@ def conversion_info(subject, outdir, info, filegroup, ses):
 
 def prep_conversion(sid, dicoms, outdir, heuristic, converter, anon_sid,
                    anon_outdir, with_prov, ses, bids, seqinfo, min_meta,
-                   overwrite, dcmconfig):
+                   overwrite, dcmconfig, bids_options):
```

Comment:
that is what I was hoping to avoid -- needing to eventually add/adjust signature of every function which now gets `bids` but might eventually need to get access to  `bids_options`...

but may be making it more obvious at the cost of some duplication would be better than my more obscure suggestion... Let's see what others thing

Reply:
Hmm I see what you're saying. I wanted to keep bids as a boolean because I think it's more obvious to people and prevents us having to check every usage of args.bids. (e.g., if bids is an empty list, we would want to calculate bids stuff, but `if args.bids:` would no longer work.) Therefore a change is required in either case, and I feel the `bids` + `bids_options` method is more explicit and clear. I'm curious what others thing.

Reply:
Yeah, I'd like to avoid additional parameters to the signatures whenever possible. Though if we're changing the variable type + intention, it may be worthwhile to rename it to something more readable - I like `bids_options` or `bids_opts`. Most of the current checks can be ported rather easily
```python
if args.bids_opts is not None:  # if args.bids
   if ['notopdirs'] in args.bids_opts:  # special case
       ...
```

Reply:
Refactored to use bids_options internally